app.js

```diff
@@ -0,0 +1,104 @@
+// dependencies
+const   express     = require('express'),
+        bodyParser  = require('body-parser'),
+        session     = require('express-session'),
+        mongoose    = require('mongoose'),
+        app         = express();
+
+// setting body parser
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: true}));
+
+// setting the home directory
+const publicDir = require('path').join(__dirname,'/public');
+app.use(express.static(publicDir));
+
+// setup for creating session and other related stuff with authentication
+app.use(session({
+    secret: 'fldsk2232',
+    resave: true,
+    saveUninitialized: true
+}))
+
+let auth = function(req,res,next){
+    if(req.session && req.session.password == password)
+        return next();      // if a valid session is already running then continue 
+    
+    else
+        return res.redirect('/');       // otherwise redirect to login form page
+}
+
+const password = "abc123"; 
+
+
+// setting up db
+mongoose.Promise = global.Promise;
+mongoose.connect('mongodb://localhost/webinar', {useNewUrlParser: true});
+mongoose.connection
+    .once('open', () => console.log('connected to database'))
+    .on('error', (error) => console.log('connection to database failed. Error = ', error));
+
+// ---------------------    MODELS   ---------------------
+
+const Webinar = require('./models/webinar');
+
+
+// ---------------------    ROUTES    ---------------------
+
+app.get('/', (req,res) => {
+    // render the login form
+    // res.send('login page')
+    res.render('login.ejs')
+})
+
+app.post('/', (req,res) => {
+    console.log(req.session);
+    let enteredPassword = req.body.password;
+    console.log(enteredPassword)
+    if(password == enteredPassword){
+        console.log('logged in');
+        req.session.password = enteredPassword;
```

Comment:
Don't save the password in the session. Save only the user ID. Or in this case, a boolean saying logged in or not